src/internal/animate.ts

```diff
@@ -45,10 +45,29 @@ export function stopAnimations(el: HTMLElement) {
   return Promise.all(
     el.getAnimations().map(animation => {
       return new Promise(resolve => {
-        const handleAnimationEvent = requestAnimationFrame(resolve);
+        const requestId = requestAnimationFrame(timestamp => {
+          animation.oncancel = null;
+          animation.onfinish = null;
+          resolve(timestamp);
+        });
 
-        animation.addEventListener('cancel', () => handleAnimationEvent, { once: true });
-        animation.addEventListener('finish', () => handleAnimationEvent, { once: true });
+        animation.addEventListener(
+          'cancel',
+          () => {
+            cancelAnimationFrame(requestId);
```

Comment:
I'm not following why we're canceling the `requestAnimationFrame()` created above. I don't see how it's related to the animation. At this point, we just need to wait for one more frame before resolving.

Reply:
Here I am trying to cancel requested animation frame for we no longer need to run the callback passed to it since we're resolving the promise here. (in the case where this handler is called before the end of current frame);